ladybug/dt.py

```diff
@@ -220,3 +224,199 @@ def ToString(self):
+
+        Args:
+            data: {
+                'month': A value for month between 1-12. (Defualt: 1)
+                'day': A value for day between 1-31. (Defualt: 1)
+            }
+        """
+        month = data['month'] if 'month' in data else 1
+        day = data['day'] if 'day' in data else 1
+        leap_year = data['leap_year'] if 'leap_year' in data else False
+        return cls(month, day, leap_year)
+
+    @classmethod
+    def from_doy(cls, doy, leap_year=False):
+        """Create Ladybug Date from an day of the year.
+
+        Args:
+            doy: An int value 0 <= and < 366
+        """
+        if not leap_year:
+            days_until_month = (0, 31, 59, 90, 120, 151, 181,
+                                212, 243, 273, 304, 334, 365)
+        else:
+            days_until_month = (0, 31, 60, 91, 121, 152, 182,
+                                213, 244, 274, 305, 335, 366)
+
+        for month_count in range(12):  # find month
+            if int(doy) < days_until_month[month_count + 1]:
+                month = month_count + 1
+                break
+        try:
+            day = int(doy - days_until_month[month - 1])
+        except UnboundLocalError:
+            raise ValueError(
+                "doy must be positive and smaller than 366. Invalid input %d" % (doy)
+            )
+        return cls(month, day, leap_year)
+
+    @classmethod
+    def from_date_string(cls, date_string, leap_year=False):
+        """Create Ladybug Date from a Date string.
+
+        Usage:
+
+            dt = Date.from_date_string("31 Dec")
+        """
+        dt = datetime.strptime(date_string, '%d %b')
+        return cls(dt.month, dt.day, leap_year)
+
+    @property
+    def leap_year(self):
+        """Boolean to note whether Date belongs to a leap year or not."""
+        return self.year == 2016
+
+    @property
+    def doy(self):
+        """Calculate day of the year for this date."""
+        return self.timetuple().tm_yday
+
+    def to_dict(self):
+        """Get date as a dictionary."""
+        base = {'month': self.month, 'day': self.day}
+        if self.leap_year:
+            base['leap_year'] = True
+        return base
+
+    def __str__(self):
+        """Return date as a string."""
+        return self.strftime('%d %b')
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__str__()
+
+    def __repr__(self):
+        """Return date as a string."""
+        return self.__str__()
+
+
+class Time(time):
+    """Create Ladybug Time.
+
+    Properties:
+        hour
+        minute
+        mod
+        float_hour
+    """
+
+    __slots__ = ()
+
+    def __new__(cls, hour=0, minute=0):
+        """Create Ladybug Time.
+
+        Args:
+            hour: A value for hour between 0-23 (Defualt: 0).
+            minute: A value for month between 0-59 (Defualt: 0).
+        """
+        hour, minute = DateTime._calculate_hour_and_minute(hour + minute / 60.0)
```

Comment:
Probably it is more natural to have this method in Time class and reference it in DateTime not that other way around.

Reply:
Good point and I agree